lib/dues_calculator.rb

```diff
@@ -24,10 +24,12 @@ def self.dues_per_competitor_in_usd(country_iso2, base_entry_fee_lowest_denomina
     DuesCalculator.update_exchange_rates_if_needed
     input_money_us_dollars = Money.new(base_entry_fee_lowest_denomination, currency_code).exchange_to("USD")
 
-    registration_fee_dues_us_dollars = input_money_us_dollars * CountryBand.percent_registration_fee_used_for_due_amount(country_band)
-    country_band_dues_us_dollars = country_band.present? && country_band > 0 ? CountryBand::BANDS[country_band][:value] : 0
-    # times 100 because Money require lowest currency subunit, which is cents for USD
-    country_band_dues_us_dollars_money = Money.new(country_band_dues_us_dollars * 100, "USD")
+    country_band_detail = CountryBandDetail.find_by(number: country_band)
+    return nil unless country_band_detail
```

Comment:
Is our code prepared to handle a `nil` return value? The previous implementation was returning `0` If the `country_band.present?` check failed.

Reply:
Also, you can check `country_band_detail.present?`. It's a bit more readable than this "hacky" null check.